src/pages/LogInWithShortLivedTokenPage.js

```diff
@@ -42,47 +45,61 @@ const propTypes = {
 };
 
 const defaultProps = {
+    isOnyxDoneClearing: false,
     route: {
         params: {},
     },
     session: {},
 };
 
 class LogInWithShortLivedTokenPage extends Component {
-    componentDidMount() {
-        const email = lodashGet(this.props.route.params, 'email', '');
-        const shortLivedToken = lodashGet(this.props.route.params, 'shortLivedToken', '');
+    constructor(props) {
+        super(props);
+        this.state = {
+            isSigningIn: false,
+        };
+    }
 
-        const isUserSignedIn = this.props.session && this.props.session.authToken;
-        if (!isUserSignedIn) {
-            Log.info('[LoginWithShortLivedTokenPage] User not signed in - signing in with short lived token');
-            Session.signInWithShortLivedToken(email, shortLivedToken);
+    componentDidMount() {
+        // We need to wait for Onyx to finish clearing before signing in because
+        // it will re-initialize with the initialKeyStates, which can notify
+        // subscribers with an out of date value causing the user to be signed
+        // out again. See https://github.com/Expensify/react-native-onyx/issues/125
```

Comment:
**thought:** This comment should be removed as it provides a lot of specific context that doesn't quite make sense unless you have a lot of context on your specific view of what's happening here and the issue. 

Separately, we are linking to a bug report which implies we should be fixing an issue in Onyx. Left a comment on that issue and think we should give it more thought.

**suggestion:** If we want to say anything here about why we are waiting for Onyx to finish clearing we could say something like:

>If we are signing an old user out then we'll want to wait until they are fully signed out and Onyx has reinitialized as a blank slate before signing in the next user.